onnxruntime/core/mlas/inc/mlas.h

```diff
@@ -207,23 +207,77 @@ MlasGemm(
     MLAS_THREADPOOL* ThreadPool
     );
 
-void
-MLASCALL
-MlasGemm(
-    size_t M,
-    size_t N,
-    size_t K,
-    const uint8_t* A,
-    size_t lda,
-    uint8_t offa,
-    const uint8_t* B,
-    size_t ldb,
-    uint8_t offb,
-    bool BIsSigned,
-    int32_t* C,
-    size_t ldc,
-    MLAS_THREADPOOL* ThreadPool
-    );
+enum class MLAS_QUANTIZATION_GRANULARITY {
+    PerMatrix,
+    PerColumn,
+};
+
+enum class MLAS_QGEMM_OUTPUT_MODE {
+    ZeroMode,       // overwrite the output buffer
+    AccumulateMode, // accumulate to the output buffer
+};
+
+class MLAS_QGEMM_OUTPUT_PROCESSOR {
+public:
+    inline
+    virtual
+    void
+    Process(
+        const int32_t* /*CBuffer*/,
+        size_t /*RangeStartM*/,
+        size_t /*RangeStartN*/,
+        size_t /*RangeCountM*/,
+        size_t /*RangeCountN*/,
+        size_t /*ldc*/
+        ) const
+    {
+        return;
+    }
+};
+
+class MLAS_QGEMM_SCALE_BIAS_OUTPUT_PROCESSOR : public MLAS_QGEMM_OUTPUT_PROCESSOR {
+public:
+    MLAS_QGEMM_SCALE_BIAS_OUTPUT_PROCESSOR(
+        float* Output,
+        size_t LeadingDimensionOutput,
+        const float* Scale,
+        const float* Bias,
+        MLAS_QGEMM_OUTPUT_MODE Mode = MLAS_QGEMM_OUTPUT_MODE::ZeroMode,
+        MLAS_QUANTIZATION_GRANULARITY QuantGran = MLAS_QUANTIZATION_GRANULARITY::PerMatrix
+        );
+
+    inline
+    void
+    Process(
+        const int32_t* CBuffer,
+        size_t StartM,
+        size_t StartN,
+        size_t CountM,
+        size_t CountN,
+        size_t ldc
+        ) const override;
+
+private:
+    template<bool HASBIAS, MLAS_QGEMM_OUTPUT_MODE MODE, MLAS_QUANTIZATION_GRANULARITY QUANTGRAN>
```

Comment:
Can this be HasBias, Mode, QuantGran instead? More consistent with existing style.